benchmark/requirements.txt

```diff
@@ -0,0 +1,3 @@
+numpy
+perfplot
+gprof2dot
```

Comment:
Are these propagated to pennylane core? What is required to "install" the benchmarking suite?

Reply:
When I run the example scripts, I get pip warnings. Where is `pip` being invoked?

Reply:
`qml.about` has some hacky code that uses `pip` internals to show PennyLane packaging information. It's very convenient, but not good from a robustness viewpoint.

Reply:
These requirements are just for the benchmarking suite. They are not absolutely necessary, perfplot is only needed if you want to do performance plots, and gprof2dot if you want to visualize profiler output.

Reply:
so does `requirements.txt` do anything programmatically? If it's just so users will know what to install, better to put that in readme I think

Reply:
I added a section in README. The requirements file is for `pip`:
```
pip install -r requirements.txt
```

Reply:
thanks! I had tried `pip install -e .` but that didn't work